src/api/app/models/bs_request.rb

```diff
@@ -1174,12 +1163,39 @@ def update_cache
 
   private
 
+  def raisepriority(new_priority)
+    # rails enums do not support compare and break db constraints :/
+    self.priority = new_priority if change_priorities?(new_priority)
+  end
+
+  # We can only raise the priority, in the context where this method is needed.
+  # This method checks makes sure this is the case.
+  def change_priorities?(new_priority)
+    new_priority == "critical" ||
+      new_priority == "important" && priority.in?(["moderate", "low"]) ||
+      new_priority == "moderate" && "low" == priority
+  end
+
   def check_bs_request_actions!(opts = {})
     bs_request_actions.each do |action|
       action.check_action_permission!(opts[:skip_source])
       action.check_for_expand_errors!(!@addrevision.nil?)
       raisepriority(action.minimum_priority)
     end
+
+    return unless persisted?
+
+    # rubocop:disable Style/GuardClause
+    if priority_changed?
```

Comment:
Why you disable rubocop here?

Reply:
Because I am not going to write a guard claus for a 7 line statement :rofl: